src/game.cpp

```diff
@@ -12013,7 +12013,7 @@ bool game::ramp_move( const tripoint &dest_loc )
 
     // We're moving from a ramp onto an obstacle
     if( !m.has_flag( TFLAG_RAMP, u.pos() ) ||
-        m.move_cost_ter_furn( dest_loc ) > 0 ) {
+        m.impassable( dest_loc ) ) {
```

Comment:
The logic here is inverted. Could be caused by my comment, which also has inverted logic (that is, the `if` checks the opposite of what comment describes).

Intended meaning is:
When trying to move from a ramp-less tile OR onto a passable tile, CANCEL the ramp move.

The inverted logic makes it so that moving from ramp onto grass on the same z-level will make survivor walk upwards (and fall), but trying to climb stairs will not work.